ddtrace/llmobs/_integrations/openai_agents.py

```diff
@@ -0,0 +1,347 @@
+import json
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Union
+
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils.formats import format_trace_id
+from ddtrace.llmobs._constants import INPUT_MESSAGES
+from ddtrace.llmobs._constants import INPUT_VALUE
+from ddtrace.llmobs._constants import METADATA
+from ddtrace.llmobs._constants import METRICS
+from ddtrace.llmobs._constants import MODEL_NAME
+from ddtrace.llmobs._constants import MODEL_PROVIDER
+from ddtrace.llmobs._constants import NAME
+from ddtrace.llmobs._constants import OUTPUT_MESSAGES
+from ddtrace.llmobs._constants import OUTPUT_VALUE
+from ddtrace.llmobs._constants import PARENT_ID_KEY
+from ddtrace.llmobs._constants import SESSION_ID
+from ddtrace.llmobs._constants import SPAN_KIND
+from ddtrace.llmobs._integrations.base import BaseLLMIntegration
+from ddtrace.llmobs._integrations.utils import LLMObsTraceInfo
+from ddtrace.llmobs._integrations.utils import OaiSpanAdapter
+from ddtrace.llmobs._integrations.utils import OaiTraceAdapter
+from ddtrace.llmobs._integrations.utils import ToolCallTracker
+from ddtrace.llmobs._utils import _get_nearest_llmobs_ancestor
+from ddtrace.llmobs._utils import _get_span_name
+from ddtrace.llmobs._utils import add_span_link
+from ddtrace.trace import Pin
+from ddtrace.trace import Span
+
+
+logger = get_logger(__name__)
+
+
+class OpenAIAgentsIntegration(BaseLLMIntegration):
+    _integration_name = "openai_agents"
+
+    def __init__(self, integration_config):
+        super().__init__(integration_config)
+        self.oai_to_llmobs_span = {}  # type: dict[str, Span]
+        self.llmobs_traces = {}  # type: dict[str, LLMObsTraceInfo]
```

Comment:
type hinting is built-in now
```suggestion
        self.oai_to_llmobs_span: Dict[str, Span] = {}  # type: dict[str, Span]
        self.llmobs_traces: Dict[str, LLMObsTraceInfo] = {}  # type: dict[str, LLMObsTraceInfo]
```